source/extensions/filters/network/ratelimit/ratelimit.h

```diff
@@ -83,13 +84,42 @@ class Filter : public Network::ReadFilter,
   void initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callbacks) override {
     filter_callbacks_ = &callbacks;
     filter_callbacks_->connection().addConnectionCallbacks(*this);
+    injectDynamicDescriptorKeys(config_->descriptors(), filter_callbacks_->connection());
+  }
+
+  void injectDynamicDescriptorKeys(std::vector<RateLimit::Descriptor> original_descriptors,
+                                   const Network::Connection& connection) {
+    std::vector<RateLimit::Descriptor> dynamicDescriptors = std::vector<RateLimit::Descriptor>();
+    for (const RateLimit::Descriptor& descriptor : original_descriptors) {
+      RateLimit::Descriptor new_descriptor;
+      for (const RateLimit::DescriptorEntry& descriptorEntry : descriptor.entries_) {
+        std::string value = descriptorEntry.value_;
+
+        // TODO: do we care about case sensitivity?
+        if (strcasecmp(descriptorEntry.key_.c_str(), "remote_address") == 0 &&
+            strcasecmp(descriptorEntry.value_.c_str(), "downstream_ip") == 0) {
```

Comment:
I think this implementation will change but no C string handling, use C++/absl/etc.